src/buildvars.py

```diff
@@ -97,17 +104,11 @@ def _force_single_ec2_node():
 
     stack_all_ec2_nodes(stackname, _force_single_ec2_node, username=BOOTSTRAP_USER)
 
-def _retrieve_build_vars():
-    print 'looking for build vars ...'
-    with hide('everything'):
-        buildvars = _validate()
-    print 'found build vars'
-    print
-    return buildvars
-
 def _update_remote_bvars(stackname, buildvars):
     LOG.info('updating %r with new vars %r', stackname, buildvars)
-    assert core_utils.hasallkeys(buildvars, ['branch'])  # , 'revision']) # we don't use 'revision'
+    # TODO: do we use 'branch' anymore? if not, remove
```

Comment:
I think not all projects have a `revision` (e.g. `api-gateway`). We may not be using `branch` anymore.